Weather binding fails to start on openHAB 2

On openHAB 2, I get the following exception:
```
23:39:46.995 [ERROR] [.weather.internal.bus.WeatherBinding] - Error activating WeatherBinding: CommonId for provider YAHOO with id 0 already exists
java.lang.IllegalArgumentException: CommonId for provider YAHOO with id 0 already exists
	at org.openhab.binding.weather.internal.parser.CommonIdHandler.addCommonId(CommonIdHandler.java:85)
	at org.openhab.binding.weather.internal.parser.CommonIdHandler.loadMapping(CommonIdHandler.java:73)
	at org.openhab.binding.weather.internal.bus.WeatherBinding.activate(WeatherBinding.java:55)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)[:1.8.0_40]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)[:1.8.0_40]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)[:1.8.0_40]
	at java.lang.reflect.Method.invoke(Method.java:497)[:1.8.0_40]
	at org.apache.felix.scr.impl.helper.BaseMethod.invokeMethod(BaseMethod.java:222)
```
The binding should allow that the activate() method is called multiple times (with deactivate inbetween of course).

The weather binding is being activated and deactivated multiple times.  The binding.xml wasn't updated with the immediate="true" fix when that was done on a bunch of bindings.  (weatherservlet.xml was updated but binding.xml wasn't).

Adding immediate="true" to the binding.xml file causes the binding to be only activated once and we don't get the duplicate id exception.

That being said I get the following lines:

```2016-01-21 12:39:35.088 [ERROR] [org.openhab.binding.weather         ] - [org.openhab.binding.weather(214)] bind method [addBindingProvider] not found; Component will fail
2016-01-21 12:39:35.097 [ERROR] [org.openhab.binding.weather         ] - [org.openhab.binding.weather(214)] bind method [addBindingProvider] not found```

And then later in the log the WeatherJobScheduler states: 
```Disabling weather locationId 'beamsville', no binding available```

More then likely they are related as the binding isn't being fully registered.

--- a/bundles/binding/org.openhab.binding.weather/OSGI-INF/binding.xml
+++ b/bundles/binding/org.openhab.binding.weather/OSGI-INF/binding.xml
@@ -9,7 +9,7 @@
     http://www.eclipse.org/legal/epl-v10.html
 
 -->
-<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.1.0" name="org.openhab.binding.weather">
+<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.1.0" immediate="true" name="org.openhab.binding.weather">
    <implementation class="org.openhab.binding.weather.internal.bus.WeatherBinding"/>
    <service>
       <provide interface="org.osgi.service.cm.ManagedService" />

--- a/bundles/binding/org.openhab.binding.weather/src/main/java/org/openhab/binding/weather/internal/bus/WeatherBinding.java
+++ b/bundles/binding/org.openhab.binding.weather/src/main/java/org/openhab/binding/weather/internal/bus/WeatherBinding.java
@@ -83,6 +83,14 @@ public class WeatherBinding extends AbstractBinding<WeatherBindingProvider>imple
         }
     }
 
+    protected void addBindingProvider(WeatherBindingProvider bindingProvider) {
+        super.addBindingProvider(bindingProvider);
+    }
+
+    protected void removeBindingProvider(WeatherBindingProvider bindingProvider) {
+        super.removeBindingProvider(bindingProvider);
+    }
+
     /**
      * Restart scheduler if all binding changes.
      */

